packages/client/webapp/src/components/Sidebar.vue

```diff
@@ -0,0 +1,134 @@
+<script setup lang="ts">
+import { ref } from 'vue';
+import Button from '@/components/Button.vue';
+import SimulationPlan from '@carbon/icons-vue/es/data-player/32';
+import MachineLearningModel from '@carbon/icons-vue/es/machine-learning-model/32';
+import Dataset from '@carbon/icons-vue/es/table--split/32';
+import ProvenanceGraph from '@carbon/icons-vue/es/flow/32';
+import Search from '@carbon/icons-vue/es/search--locate/32';
+import User from '@carbon/icons-vue/es/user/32';
+
+const props = defineProps<{ sidebarPosition: String }>();
+const emit = defineEmits(['updateSidebarPosition']);
+
+const mode = ref('Simulation Plan');
+const collapsed = ref(false);
+
+function updateMode(modeOption: string) {
+	if (modeOption === mode.value) {
+		collapsed.value = !collapsed.value;
+		return;
+	}
+	collapsed.value = false;
+	mode.value = modeOption;
+}
+
+function moveSidebar() {
+	if (localStorage.sidebarPosition === 'left') {
```

Comment:
These `left` and `right` strings should also be extracted to avoid repeating hardcoded strings all over the place.

Consider:
- If the only options we want to support for the time being are "left" and "right", we can use a boolean like `isSidebarPositionedRight` instead of a string. This if/else chain can then be simplified to `const currentPosition = localStorage.getItem('sidebarPosition'); localStorage.setItem(!currentPosition)`
- If we expect to support other positions, or prefer to be explicit about what the "non-right" position is, this is a good place to use a constant enum.